### PR TITLE
Changed share argument default in .launch method

### PR DESCRIPTION
**Pull request by EdAbati**
# Description

This addresses what discussed in #1371.

Closes: #1371

# Checklist:

- [x] I have performed a self-review of my own code
- [x] My code follows the style guidelines of this project
- [x] I have commented my code in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: gradio/blocks.py [hunk 1]
```diff
@@ -634,10 +639,12 @@ def launch(
         utils.launch_counter()
 
         # If running in a colab or not able to access localhost,
-        # automatically create a shareable link.
-        is_colab = utils.colab_check()
+        # a shareable link must be created.
         if is_colab or (_frontend and not networking.url_ok(self.local_url)):
-            share = True
+            if not self.share:
```
Comment (abidlabs) [inline gradio/blocks.py:644]: We should only raise this exception if a user explicitly sets `share=False` in colab. It looks like this exception will be raised even if they call `.launch()` without passing in an explicit value for `None`

**Comment by abidlabs** [comment]
This approach looks perfect, thanks @EdAbati!

**Comment by EdAbati** [comment]
Thanks @abidlabs for the super quick feedback :)
I will add some unit tests then

**Comment by abidlabs** [comment]
Awesome thanks!

**Comment by EdAbati** [comment]
Hi @abidlabs , I made a small change by adding a 'setter' to `Blocks.share`. I think it is a bit nicer and also easier to test.
I also added then the unit tests about that.

Let me know if I missed anything. :)

**Comment by EdAbati** [comment]
Hi @abidlabs , not sure if you saw it already but 1 test fails. This is because of `test_xray`: `demo.get_config_file` seems to be affected by the unit tests I added. 
In `XRAY_CONFIG` dictionary, the `"id"` in the `components` fields are increasing the more tests I add, for example if I comment out 1 test the `"id"` of the 1st component is `2`, with 2 tests the id of the 1st component is `3`.
I tried to run the unit tests sequentially manually and they are all successful.

Do you know why the tests are not indipendent from each other? What am I missing?

**Comment by abidlabs** [comment]
Ah interesting, good catch. This lack of independence is because there is a single global Context that manages all of the IDs that are defined in a Python session. Let me fix the test so that this is no longer an issue.

**Comment by abidlabs** [comment]
LGTM, thank you @EdAbati for the nice contribution and tests!

**Comment by EdAbati** [comment]
Thank you!

**Pull request merged**